clio-client/src/main/scala/org/broadinstitute/clio/client/commands/CommandDispatch.scala

```diff
@@ -27,32 +28,33 @@ class CommandDispatch(val webClient: ClioWebClient, val ioUtil: IoUtil)
     }
   }
 
-  def dispatch(
-    config: BaseArgs
-  )(implicit ec: ExecutionContext): Future[Boolean] = {
+  def dispatch(config: BaseArgs)(implicit ec: ExecutionContext): Future[Unit] = {
     config.command
       .map(command => execute(command, config))
-      .fold(Future(false))(checkResponse)
+      .fold(Future.failed[Unit](new Exception("The config command was empty")))(
+        checkResponse
+      )
   }
 
   def checkResponse(
     responseFuture: Future[HttpResponse]
-  )(implicit ec: ExecutionContext): Future[Boolean] = {
-    responseFuture.map { response =>
-      val isSuccess = response.status.isSuccess()
-      if (isSuccess) {
-        logger.info(
-          s"Successfully completed command." +
-            s" Response code: ${response.status}"
-        )
-      } else {
-        logger.error(
-          s"Error executing command." +
-            s" Response code: ${response.status}"
-        )
-      }
-      logger.info(response.toString)
-      isSuccess
+  )(implicit ec: ExecutionContext): Future[Unit] = {
+    responseFuture.transformWith {
+      case Success(response) =>
+        if (response.status.isSuccess()) {
+          logger.info(
+            s"Successfully completed command." +
+              s" Response code: ${response.status}"
+          )
+          Future.successful(())
```

Comment:
It seems like this discards `response`. What if the caller needs to know if the response was a `200` or a `201`? Or what the body contains? E.g.  why not return `responseFuture` instead of `Future.successful()`?

Reply:
That's an oopsie on my part.